File: core/src/main/java/heroes/journey/entities/actions/options/OptionAction.java
package heroes.journey.entities.actions.options;

import heroes.journey.entities.actions.Action;
import heroes.journey.entities.actions.Cost;

public abstract class OptionAction extends Action {
    private String display;

    public OptionAction(
        String id,
        String name,
        String description,
        boolean returnsActionList,
        Cost cost,
        String display) {
        super(id, name, description, returnsActionList, cost);
        this.display = display != null ? display : "";
    }
    
    public void setDisplay(String value) {
        this.display = super.toString() + ": " + value;
    }
}

File: core/src/main/java/heroes/journey/entities/actions/options/OptionAction.kt
package heroes.journey.entities.actions.options

import heroes.journey.entities.actions.Action
import heroes.journey.modlib.actions.ActionResult
import heroes.journey.modlib.actions.IActionContext
import heroes.journey.modlib.actions.IOptionAction
import heroes.journey.modlib.actions.ShowAction
import heroes.journey.modlib.attributes.Attributes

abstract class OptionAction(
    id: String,
    requirementsMetFn: (IActionContext) -> ShowAction = { ShowAction.YES },
    onHoverFn: (IActionContext) -> Unit = {},
    onSelectFn: (IActionContext) -> ActionResult,
    override var value: Any,
    cost: Attributes? = null
) : Action(
    id,
    false,
    requirementsMetFn,
    onHoverFn,
    onSelectFn,
    cost = cost
), IOptionAction {
    override fun getName(): String {
        return super.getName() + ": " + value
    }
}
